From: Mauro Carvalho Chehab <mchehab@kernel.org>
Subject: mm: document maple tree pointer at unmap_vmas() at memory.c
Date: Tue, 28 Jun 2022 10:46:19 +0100

A maple tree pointer parameter was added at unmap_vmas() without
description:

	mm/memory.c:1729: warning: Function parameter or member 'mt' not described in 'unmap_vmas'

Document it.

Link: https://lkml.kernel.org/r/d2690118414c5458c448348cc94f831d8e8444ce.1656409369.git.mchehab@kernel.org
Fixes: 5966e5c968b2 ("mm: remove the vma linked list")
Fixes: f8acc5e9581e ("Maple Tree: add new data structure")
Signed-off-by: Mauro Carvalho Chehab <mchehab@kernel.org>
Cc: "Liam R. Howlett" <Liam.Howlett@Oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/memory.c~mm-remove-the-vma-linked-list-fix-2
+++ a/mm/memory.c
@@ -1699,6 +1699,8 @@ static void unmap_single_vma(struct mmu_
 /**
  * unmap_vmas - unmap a range of memory covered by a list of vma's
  * @tlb: address of the caller's struct mmu_gather
+ * @mt: maple tree pointer. A maple tree is a RCU-safe range based B-tree
+ *	designed to use modern processor cache efficiently
  * @vma: the starting vma
  * @start_addr: virtual address at which to start unmapping
  * @end_addr: virtual address at which to end unmapping
_
